core/launch_configuration_test.go

```diff
@@ -104,3 +105,265 @@ func Test_copyBlockDeviceMappings(t *testing.T) {
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
+					Arn: aws.String("arn:aws:something"),
+				},
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
+		{
+			name: "IAM instame profile name",
+			lc: &launchConfiguration{
+				&autoscaling.LaunchConfiguration{
+					IamInstanceProfile: aws.String("bla bla bla something"),
+				},
+			},
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
+					Name: aws.String("bla bla bla something"),
+				},
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
+		{
+			name: "IAM instame profile name",
+			lc: &launchConfiguration{
+				&autoscaling.LaunchConfiguration{
+					KeyName: aws.String("key xyz"),
+				},
+			},
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				KeyName:      aws.String("key xyz"),
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
+		{
+			name: "instance monitoring",
+			lc: &launchConfiguration{
+				&autoscaling.LaunchConfiguration{
+					InstanceMonitoring: &autoscaling.InstanceMonitoring{
+						Enabled: aws.Bool(false),
+					},
+				},
+			},
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				Monitoring: &ec2.RunInstancesMonitoringEnabled{
+					Enabled: aws.Bool(false),
+				},
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
+		{
+			name: "user data",
+			lc: &launchConfiguration{
+				&autoscaling.LaunchConfiguration{
+					UserData: aws.String("user data"),
+				},
+			},
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				UserData:     aws.String("user data"),
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
+		{
+			name: "networking",
+			lc: &launchConfiguration{
+				&autoscaling.LaunchConfiguration{
+					AssociatePublicIpAddress: aws.Bool(true),
+				},
+			},
+			instance: &instance{
+				Instance: &ec2.Instance{},
+			},
+			spotRequest: &ec2.RequestSpotLaunchSpecification{
+				NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
+					{
+						AssociatePublicIpAddress: aws.Bool(true),
+						DeviceIndex:              aws.Int64(0),
+					},
+				},
+				InstanceType: aws.String(""),
+				Placement: &ec2.SpotPlacement{
+					AvailabilityZone: aws.String(""),
+				},
+			},
+		},
```

Comment:
Maybe adding a test with multiple parameter: az/network/storage/user-data/name etc, could be good in order to validate at least one of these cases.